internal/cmd/commands/server/server.go

```diff
@@ -469,6 +469,12 @@ func (c *Command) Run(args []string) int {
 			c.Info["worker auth registration request"] = c.worker.WorkerAuthRegistrationRequest
 			c.InfoKeys = append(c.InfoKeys, "worker auth current key id")
 			c.Info["worker auth current key id"] = c.worker.WorkerAuthCurrentKeyId
+
+			// Write the WorkerAuth request to a file
+			if err := c.StoreWorkerAuthReq(c.worker.WorkerAuthRegistrationRequest, c.worker.WorkerAuthStorage.BaseDir()); err != nil {
+				c.UI.Error(fmt.Errorf("Error storing worker auth request: %w", err).Error())
```

Comment:
This needs the same treatment as in dev.go -- if it fails, shut down the worker and then the controller before exiting.